invisible_cities/detector_simulation/buffer_functions_test.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+
+from pytest import mark
+
+from .. core            import system_of_units as units
+
+from . buffer_functions import       bin_sensors
+from . buffer_functions import calculate_buffers
+from . buffer_functions import       find_signal
+
+
+def test_bin_sensors(mc_waveforms, pmt_ids, sipm_ids):
+    max_buffer = 10 * units.minute
+
+    evts, pmt_binwid, sipm_binwid, all_wfs = mc_waveforms
+
+    evt   = evts[0]
+    wfs   = all_wfs.loc[evt]
+
+    pmts  = wfs[wfs.index.isin( pmt_ids)]
+    sipms = wfs[wfs.index.isin(sipm_ids)]
+
+    ## Assumes pmts the triggering sensors as in new/next-100
+    bins_min = pmts.time.min()
+    bins_max = pmts.time.max() + pmt_binwid
+    pmt_bins ,  pmt_wf = bin_sensors(pmts ,  pmt_binwid,
+                                     bins_min, bins_max, max_buffer)
+    sipm_bins, sipm_wf = bin_sensors(sipms, sipm_binwid,
+                                     bins_min, bins_max, max_buffer)
+
+    assert np.all(np.diff( pmt_bins) ==  pmt_binwid)
+    assert np.all(np.diff(sipm_bins) == sipm_binwid)
+    assert pmt_bins[ 0] >= sipm_bins[ 0]
+    assert pmt_bins[-1] >= sipm_bins[-1]
+
+    pmt_sum  = pmts .charge.sum()
+    sipm_sum = sipms.charge.sum()
+    assert pmt_wf .sum().sum() ==  pmt_sum
+    assert sipm_wf.sum().sum() == sipm_sum
+
+
+@mark.parametrize("signal_thresh", (2, 10))
+def test_find_signal(binned_waveforms, signal_thresh):
+
+    pmt_bins, pmt_wf, *_ = binned_waveforms
+
+    buffer_length = 800
+    bin_width     = np.diff(pmt_bins)[0]
+    stand_off     = int(buffer_length * units.mus / bin_width)
+
+    pmt_sum       = pmt_wf.sum()
+    pulses        = find_signal(pmt_wf, signal_thresh, stand_off)
+
+    assert np.all(pmt_sum[pulses] > signal_thresh)
+
+
+@mark.parametrize("pre_trigger signal_thresh".split(),
+                  ((100,  2),
+                   (400, 10)))
+def test_calculate_buffers(mc_waveforms, binned_waveforms,
+                           pre_trigger ,    signal_thresh):
+
+    _, pmt_binwid, sipm_binwid, _ = mc_waveforms
+
+    pmt_bins, pmt_wf, sipm_bins, sipm_wf = binned_waveforms
+
+    buffer_length     = 800
+    bin_width         = np.diff(pmt_bins)[0]
+    stand_off         = int(buffer_length * units.mus / bin_width)
+
+    pulses            = find_signal(pmt_wf, signal_thresh, stand_off)
+
+    buffer_calculator = calculate_buffers(buffer_length,
+                                          pre_trigger  ,
+                                          pmt_binwid   ,
+                                          sipm_binwid  )
+
+    buffers           = buffer_calculator(pulses, *binned_waveforms)
+    pmt_sum           = pmt_wf.sum()
+
+    assert len(buffers) == len(pulses)
+    for i, evt in enumerate(buffers):
```

Comment:
```suggestion
    for i, (evt_pmt, evt_sipm) in enumerate(buffers):
```
to avoid `evt[0 / 1]` later